docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+      )
+  ))
+  (contract %callback
+    (pair
+      (pair %operator
+        (address %owner)
+        (pair
+          (address %operator)
+          (or %tokens
+            (unit %allTokens)
+            (set %someTokens nat)
+          )
+      ))
+      (bool %isOperator)
+    )
+  )
+)
+```
+
+- Inspect if an address is an operator for the specified owner and    token types.
+
+- If the adddress is not an operator for at least one
+  requested token type, then the result is `false`.
+
+- It's possible to make this query for some specific tokens, or to
+  all tokens.
+
+**Managed Ledger Functions (TBD)**
+----------------------------
+
+TODO: Maybe it's better to omit these mentions since all that left from
+ManagedLedger is their semantics?
+
+Functions for the TZUSDC token implementation found in Tezos [*ManagedLedger*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-7/ManagedLedger.md)
+
+**pause**
+
+Types
+```
+pause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %pause
+  (address %sender)
+  unit
+)
+```
+
+- Pauses transferring, burning, minting and allowance operations so
+  that they cannot be performed. All other operations remain
+  unaffected.
+
+- Cannot be called multiple times.
+
+- Sender must be a pauser.
+
+**unpause**
+
+Types
+```
+unpause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %unpause
+  (address %sender)
+  unit
+)
+```
+
+- Unpauses the contract so that transferring, burning, minting and
+  allowance operations can be performed by users with corresponding
+  roles.
+
+- Cannot be called multiple times.
+
+- Sender must be a pauser.
+
+**getPaused**
+
+Types
+```
+tokenId = nat
+
+getPausedResponse =
+  ( tokenId :tokenId
+  , nat     :paused
+  )
+
+getPausedParam =
+  ( list tokenId                      :tokenIds
+  , contract (list getPausedResponse) :callback
+  )
+
+getPaused = getPausedParam
+```
+
+Parameter (in Michelson)
+```
+(pair %getTotalPaused
+  (list %tokenIds nat)
+  (contract %callback
+    (list
+      (pair %getPausedResponse
+        (nat %tokenId)
+        (bool %paused)
+      )
+    )
+  )
+)
+```
+
+- Returns token pause status.
+
+- *Note: It's not a part of ManagedLedger interface*.
+
+**Custom token functions**
+==========================
+
+**Issuing and destroyng tokens**
+--------------------------
+
+**configureMinter**
+
+Types
+```
+tokenId = nat
+
+configureMinterParam =
+  ( address :sender
+  , address :minter
+  , tokenId :tokenId
+  )
+
+configureMinter = configureMinterParam
+```
+
+Parameter (in Michelson)
+```
+(pair %configureMinter
+  (address %sender)
+  (pair
+    (address %minter)
+    (nat %tokenId)
+))
+```
+
+- Adds `minter` to sender's minter list to allow him mint tokens of specific
+  token type.
+
+- Sender must master minter.
+
+- Minter cannot mint tokens untill `setMintingAllowance`
+  is specified.
+
+- Cannot add minter if present in someone else's minter list.
```

Comment:
Implementation hint: it's better to have a global set (or, rather, `big_map address ()`) of all minters so that you can check this quickly without iterating over master minters' minters lists.